server/odc-service/src/main/java/com/oceanbase/odc/service/task/executor/task/SqlPlanTask.java

```diff
@@ -0,0 +1,433 @@
+import com.oceanbase.odc.core.shared.exception.InternalServerError;
+import com.oceanbase.odc.core.shared.exception.UnexpectedException;
+import com.oceanbase.odc.core.sql.execute.SyncJdbcExecutor;
+import com.oceanbase.odc.core.sql.execute.model.JdbcGeneralResult;
+import com.oceanbase.odc.core.sql.execute.model.SqlExecuteStatus;
+import com.oceanbase.odc.core.sql.execute.model.SqlTuple;
+import com.oceanbase.odc.core.sql.parser.AbstractSyntaxTreeFactories;
+import com.oceanbase.odc.core.sql.parser.AbstractSyntaxTreeFactory;
+import com.oceanbase.odc.core.sql.split.OffsetString;
+import com.oceanbase.odc.core.sql.split.SqlCommentProcessor;
+import com.oceanbase.odc.service.common.FileManager;
+import com.oceanbase.odc.service.common.model.FileBucket;
+import com.oceanbase.odc.service.common.util.OdcFileUtil;
+import com.oceanbase.odc.service.connection.model.ConnectionConfig;
+import com.oceanbase.odc.service.flow.task.util.TaskDownloadUrlsProvider;
+import com.oceanbase.odc.service.objectstorage.cloud.CloudObjectStorageService;
+import com.oceanbase.odc.service.schedule.job.SqlPlanJobReq;
+import com.oceanbase.odc.service.session.OdcStatementCallBack;
+import com.oceanbase.odc.service.session.factory.DefaultConnectSessionFactory;
+import com.oceanbase.odc.service.session.model.SqlExecuteResult;
+import com.oceanbase.odc.service.sqlplan.model.SqlPlanTaskResult;
+import com.oceanbase.odc.service.task.caller.JobContext;
+import com.oceanbase.odc.service.task.constants.JobParametersKeyConstants;
+import com.oceanbase.odc.service.task.util.JobUtils;
+import com.oceanbase.tools.dbbrowser.parser.ParserUtil;
+import com.oceanbase.tools.dbbrowser.parser.constant.GeneralSqlType;
+
+import lombok.Data;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class SqlPlanTask extends BaseTask<SqlPlanTaskResult> {
+
+    private SqlPlanJobReq parameters;
+
+    private ConnectionSession connectionSession;
+
+    private SyncJdbcExecutor executor;
+
+    private SqlPlanTaskResult result;
+
+    private volatile boolean canceled = false;
+
+    private File resultJsonFile;
+
+    private String zipFileRootPath;
+
+    private String errorRecordPath = null;
+
+    private final List<SqlExecuteResult> queryResultSetBuffer = new ArrayList<>();
+
+    private final List<CSVExecuteResult> csvFileMappers = new ArrayList<>();
+
+    @Override
+    protected void doInit(JobContext context) {
+        this.result = new SqlPlanTaskResult();
+        this.parameters = JobUtils.fromJson(getJobParameters().get(JobParametersKeyConstants.META_TASK_PARAMETER_JSON),
+                SqlPlanJobReq.class);
+        this.connectionSession = generateSession();
+        this.executor = connectionSession.getSyncJdbcExecutor(ConnectionSessionConstants.CONSOLE_DS_KEY);
+        initFile();
+    }
+
+    @Override
+    protected boolean doStart(JobContext context) throws Exception {
+        String sqlContent = parameters.getSqlContent();
```

Comment:
What if the SQL is in file? Does OBCloud support uploading SQL scripts?

Reply:
I have labeled this function todo and will add it in the next step.